src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2596,15 +2746,33 @@ namespace winrt::TerminalApp::implementation
         // TermControl will copy the settings out of the settings passed to it.
 
         const auto content = _manager.CreateCore(settings.DefaultSettings(), settings.UnfocusedSettings(), connection);
+        return _InitControl(TermControl{ content });
+    }
 
-        TermControl term{ content };
+    TermControl TerminalPage::_InitControlFromContent(const uint64_t& contentId)
+    {
+        if (const auto& content{ _manager.TryLookupCore(contentId) })
+        {
+            // We have to pass in our current keybindings, because that's an
+            // object that belongs to this TerminalPage, on this thread. If we
+            // don't, then when we move the content to another thread, and it
+            // tries to handle a key, it'll callback on the original page's
+            // stack, inevitably resulting in a wrong_thread
+            return _InitControl(TermControl::AttachContent(content, *_bindings));
+        }
+        return nullptr;
+    }
 
+    TermControl TerminalPage::_InitControl(const TermControl& term)
```

Comment:
nit: overload on the word `Init`. Is there really nothing more descriptive? _Attach_?